web/views/shared_view.ex

```diff
@@ -110,7 +110,7 @@ defmodule CoursePlanner.SharedView do
   end
 
   def form_date(form, field, opts \\ []) do
-    value = opts[:value] || Date.utc_today()
+    value = Map.get(form.source.changes, field) || opts[:value] || Ecto.Date.utc()
```

Comment:
inside the SharedView we should be able to call
```elixir
input_value(form, field)
```
instead of
```elixir
Phoenix.HTML.Form.input_value(form, field)
```
though